\chapter{Background}
\label{cha:intro}



\section{Liquid biopsy}
In molecular diagnosis, liquid biopsy is an attractive field to both researchers and commercial organizations, because of its non-invasive nature. RNA markers have several advantages over DNA and protein biomarkers, including its higher sensitivity, tissue specificity and lower cost for detection(~\cite{xi2017rna}). exRNAs (extracellular RNA), defined as RNAs that can be secreted outside of the cell, which make up the majority of detectable RNA species in body fluid, is currently under intensive study.

\section{exRNA as biomarkers}
Several works suggested that some exRNAs are functionally active, and may play vital roles in communication between cells under both physiological and pathological conditions(~\cite{nedaeinia2017circulating}). In mammalian cells, intracellular RNAs are secreted into body fluids in various forms: bound by proteins and form RNPs complex, or wrapped in membrane structure like microvesicles or exosomes(~\cite{nedaeinia2017circulating}). miRNA, piRNA, fragments of mRNA, snRNA, lncRNA, tRNA and Y RNA have been detected in exRNAs(~\cite{yuan2016plasma}). Most of the published studies on exRNA biomarker discovery focus on miRNA, as cancers are often accompanied by miRNA dysregulation, and miRNAs are relatively stable in body fluid(~\cite{mitchell2008circulating}). Extracellular miRNAs are reported to be capable of predicting varies cancers(~\cite{nedaeinia2017circulating}, ~\cite{wozniak2015circulating}, ~\cite{singh2014serum}, ~\cite{schwarzenbach2014clinical}, (~\cite{zhou2011plasma})). Recently, the clinical value of other RNA types has also been recognized(~\cite{li2019extracellular}). For example, several lncRNAs in exRNA were identified as biomarkers for cancers. 
Deep sequencing makes it possible to monitor RNA fragments in the sample simultaneously without prior knowledge of their sequences. exRNA sequencing is becoming a powerful tool for novel biomarker discovery, although many technical problems in both experiment and data processing remain to be solved. RNA concentration in body fluid is much lower than that in the tissue sample, lead to pervasive dropouts in exRNA sequencing data(~\cite{laurent2015meeting}). Exogenous RNAs are often present in exRNAs, which may interfere assignment of the reads(~\cite{kaczor2017novel}, ~\cite{beatty2014small}, ~\cite{allen2018bioinformatic}). The short length of exRNA also increase the difficulty of correct mapping, and many small RNA sequencing analyzing tools have to alleviate such problem via mapping the reads to different types of RNA in a sequential manner(~\cite{wu2017srnanalyzer}, ~\cite{baras2015mirge}). Different exRNA isolation protocols are biased toward certain types of exRNA cargos(~\cite{srinivasan2019small}, ~\cite{murillo2019exrna}). Considerable heterogeneity and batch effect can often be observed in exRNA sequencing data, as standard exRNA sequencing protocol still have not been established(~\cite{laurent2015meeting}, ~\cite{li2015comparison}). 
exRNAs are highly fragmented. Most of exRNAs, except for miRNA and piRNA, generally exist as small fragments, instead of annotated full-length sequences(~\cite{wei2017coding}). Previous studies suggested some of these small RNA fragments are not residues of random degradation. A fraction of small RNA fragments, with their counts significantly higher than the background(~\cite{allen2018bioinformatic}, ~\cite{wei2017coding}), can be mapped to certain secondary structure, or certain regions in longer transcripts. It has been suggested that some small RNA fragments are derived from longer noncoding RNAs like tRNA, YRNA, vault RNA, circular RNA, snRNA, and snoRNA via specific molecular machinery, and may carry out functions distinct from their parents. Alternatively, only fragments with certain secondary structures or protein binding sites could survive from RNAase degradation. If some of the full length annotated RNAs are adopted as biomarkers, more clinical applicable methods other than deep sequencing, like PCR, may lead to the inconsistent outcome. Compared to full-length tRNA, rRNA, and snRNA, fragments derived from these RNAs perform better in distinguishing exRNA from different body fluid or different subcomponents of exRNAs in same body fluid. Hence, directly utilize recurring small RNA fragments as biomarker candidates seem to be a better choice.

\section{Previous Study}
Several bioinformatics tools have been developed for the analysis of small RNA sequencing data. Chimira, Oasis(~\cite{capece2015oasis}), miRge and sRNAnalyzer are general software for small RNA sequencing data analysis, without considering the unique properties of exRNA sequencing. ExceRpt(~\cite{rozowsky2019excerpt}) and TIGER were designed for exRNA sequencing data analysis, but these two tools mainly concern the quantification of RNAs, identification of their origins, and downstream analysis like differential expression, instead of biomarker discovery. 

\section{Aim of the study}
For liquid biopsy biomarker discovery, beyond proper assignment of the reads, researchers have to go further, otherwise, biological variations that would be sufficient to distinguish cancer from health may be overwhelmed by technical variations such as the difference in sequencing depth and the existence of batch effect(~\cite{leek2010tackling}). Here we developed exSEEK, a pipeline for identifying potential biomarkers from either short of long exRNA-seq data for several cancersâ€™ classification. After the sequential mapping of the reads, we proposed a new peak calling method to explicitly find domain features of long RNAs. exSEEK can be customized to explicitly preprocess raw expression data with several methods for normalization and removing batch effect; and potential biomarkers can be robustly selected from the combination of recurring RNA fragments from long RNAs and full-length miRNA, via customizable feature selection algorithms. Our novel peak calling method for identifying structured or RBP associated domain in long RNA as features enable us to find domain features of some long RNAs including circular RNA, snoRNA, snRNA, srpRNA, etc.
  


